### ISSUE TITLE
Some modules back up files without preservation of metadata

### ISSUE THREAD

**Issue by pirat89** [upgrades-migrations/preupgrade-assistant-modules#45]
Use of the `cp` command for back up of files (usually config files) just without parameters is wrong in most cases as the context, ownership, DAC, ... are not preserved. The command should be used with `-a` or `--archive` parameter in the most cases. There is similar problem for modules written in Python.

It is important especially in case of migration. But it should be processed in the same way for upgrades too.

**Comment by AloisMahdal** [comment]
We can probably make quick-fixes here, but OTOH more efficient solution would be to replace the manual copying by API similar to [`attach_file()`][1] (probably a wrapper around that function), which already takes care of this and other issues (eg. figuring out the path, etc.).

  [1]: https://github.com/upgrades-migrations/preupgrade-assistant/pull/255

**Comment by pirat89** [comment]
Yes, that would be much more cleaner solution. Maybe just quickfix for now will be enough. There would be still some really special cases, when such solution can make still troubles. We will see. The really clean solution is to have API for work with configuration files. But this is not on the table already.
